docker-compose.yml

```diff
@@ -0,0 +1,27 @@
+version: '3'
+
+services:
+  server:
+    container_name: server 
+    build: ./server
+    image: customserver
+    ports:
+      - '3000:3000'
+    volumes:
+      - ./server/:/usr/src/app
```

Comment:
Your going to need to change these volumes to mount a tmp directory into the container, seems that when rails starts it writes a bunch of stuff to `/usr/src/app/tmp`

I'll go through how I figured this out, 

I was trying to run things locally after stopping all the docker images, and i got this error
```
$ bundle exec bin/rails
Traceback (most recent call last):
	18: from bin/rails:3:in `<main>'
...
	 1: from /home/russell/.asdf/installs/ruby/2.6.5/lib/ruby/gems/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/store.rb:83:in `dump_data'
/home/russell/.asdf/installs/ruby/2.6.5/lib/ruby/gems/2.6.0/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/store.rb:83:in `binwrite': Permission denied @ rb_sysopen - /home/russell/projects/mentor-mentee-platform/server/tmp/cache/bootsnap-load-path-cache.64211.22291.tmp (Errno::EACCES)
```

if i list the files, the owner is **root**

```
$ ls -l tmp
total 12
drwxr-xr-x 3 root root 4096 Aug 13 10:44 cache
drwxr-xr-x 2 root root 4096 Aug 13 10:45 pids
-rw-r--r-- 1 root root    0 Aug 13 10:44 restart.txt
drwxr-xr-x 2 root root 4096 Aug 13 10:44 sockets
```

This files that is written, are created by a process within the container, it writes the files to my local project directory as the root user because docker will use the root user by default.  Changing the user the app runs as is one option, but it's not a straight forward process to do that.


The problem is that, because it writes as the root user it creates a bunch of files i can't work with easily, because i can't remove the files easily without using `sudo`

The easiest option is to replace the entire `/usr/src/app/tmp/` with a tmpfs mount.  But this looks like it will effect some https://github.com/RailsGirlsCPH/mentor-mentee-platform/blob/master/server/config/environments/development.rb#L17

Perhaps that caching file could be moved?

To apply this suggestion, you'll also have to update the compose file version to **3.2**
```
version: '3.2'
```

Then adding a tmpfs
```
      - type: tmpfs
        target: /usr/src/app/tmp/
```

This seems to work ok... but has it's own limitations, since it seems to still create the `tmp` directory in the server folder, but it only does this top level folder, so you could the folder to git repo with a `.keep` file. That would prevent it being created by the root user.  But this is still a bit of a pain.

Perhaps the best solution would be to parameterise the storage of these tmp files so you can set the tmp directory to /tmp inside the Docker container using an environment variable like this https://stackoverflow.com/questions/35077463/change-temporary-directory-in-rails-4 this still seems a bit hacky to me, I'll check at work if there is a better way